xc/xc7/tests/soc/ibex/CMakeLists.txt

```diff
@@ -1,78 +1,102 @@
-add_file_target(FILE led.vmem)
-add_file_target(FILE pins_artya7.sdc)
+get_filename_component(IBEX_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../third_party/ibex ABSOLUTE)
+
+set(SV2V_ROM_INIT_FILE led.vmem)
+set(SV2V_IBEX_FLAGS --define=RegFile=ibex_pkg::RegFileFF --define=SRAMInitFile=${SV2V_ROM_INIT_FILE})
+
+# Get the list of source files used by the fusesoc to build the Ibex example
+
+execute_process(
+  COMMAND
+    bash generate.sh --only-deps
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  OUTPUT_VARIABLE FILES_FROM_GENERATE_SCRIPT
+)
 
-get_target_property_required(PYTHON3 env PYTHON3)
+# Sort the file list into sources and includes
+
+set(SOURCE_FILE_LIST "")
+set(INCLUDE_FILE_LIST "")
+
+string(REPLACE "\n" ";" FILE_LIST "${FILES_FROM_GENERATE_SCRIPT}")
+foreach(FILE_NAME ${FILE_LIST})
+  get_filename_component(FILE_EXT ${FILE_NAME} EXT)
+  if (${FILE_EXT} STREQUAL ".sv")
+    list(APPEND SOURCE_FILE_LIST ${FILE_NAME})
+  elseif (${FILE_EXT} STREQUAL ".svh")
+    list(APPEND INCLUDE_FILE_LIST ${FILE_NAME})
+  endif()
+endforeach()
+
+# Add a command for generating the Ibex source files in the build directory
+
+add_custom_command(
+  OUTPUT ${SOURCE_FILE_LIST} ${INCLUDE_FILE_LIST}
+  COMMAND
+    bash generate.sh
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  DEPENDS ${PYTHON3} ${IBEX_DIR}
+)
 
-# Arty
+# Create file targets
+
+add_file_target(FILE ${SV2V_ROM_INIT_FILE})
+get_file_target(SV2V_ROM_INIT_FILE_TARGET ${SV2V_ROM_INIT_FILE})
+
+foreach(INC ${INCLUDE_FILE_LIST})
+  add_file_target(FILE ${INC} GENERATED)
+endforeach()
+
+foreach(SRC ${SOURCE_FILE_LIST})
+  add_file_target(FILE ${SRC} GENERATED)
+  get_file_target(TARGET ${SRC})
+  add_dependencies(${TARGET} ${SV2V_ROM_INIT_FILE_TARGET})
```

Comment:
Why not just have the `SV2V_ROM_INIT_FILE` also be part the INCLUDE_FILES property?

Reply:
Done!